pettingzoo/utils/save_observation.py

```diff
@@ -1,14 +1,17 @@
 from __future__ import annotations
 
 import os
+from typing import Any
 
 import gymnasium.spaces
 import numpy as np
 
 from pettingzoo.utils.env import AECEnv, AgentID, ParallelEnv
 
 
-def _check_observation_saveable(env: AECEnv | ParallelEnv, agent: AgentID) -> None:
+def _check_observation_saveable(
+    env: AECEnv[AgentID, Any, Any] | ParallelEnv[AgentID, Any, Any], agent: AgentID
```

Comment:
Should the Any, Any not be ObsType and ActionType?

Reply:
Yes, you could write it like that, but if you don't use a generic variable in more than one place in a function definition, then it acts just like Any, since it's not actually constraining anything.

Reply:
Fair enough, I just figure it may be more readable for people to see that it's those 3 arguments even if it doesn't change the performance. Not like typing in Python really even does much anyways besides things like pyright

Reply:
When I replaced them with the typevariables, I got a warning from pyright. Apparently it's not preferred to use a TypeVar only once in a generic function signature.

Reply:
Ok whatever pyright wants is fine then